File: DSA/Shreya_Goyal_590015491/Day29Question2.c
#include <stdio.h>

int main() {
    int n, i;
    
    printf("n: ");
    scanf("%d", &n);
    
    printf("i: ");
    scanf("%d", &i);

    int bit = (n >> i) & 1;

    printf("The bit at position %d is: %d\n", i, bit);

    return 0;
}

File: DSA/VaniJain_590012020/Day29Question2.java
import java.util.*;
public class CheckBitSet 
{
    public static void main(String[] args) 
  {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter an integer (n): ");
        int n = sc.nextInt();
        System.out.print("Enter the bit position (i): ");
        int i = sc.nextInt();
        if ((n & (1 << i)) != 0)
            System.out.println("The " + i + "-th bit is SET (1).");
        else
            System.out.println("The " + i + "-th bit is NOT SET (0).");
    }
}
